resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/WorkspaceSessionTest.java

```diff
@@ -54,9 +52,27 @@ public WorkspaceSessionTest(String name) {
 		super(name);
 	}
 
+	/**
+	 * Executes the cleanup functionality in {@link ResourceTest#tearDown()} after
+	 * the last of the session of this test class (i.e., the last <code>test*</code>
+	 * method) has finished.
+	 * <p>
+	 * The test methods are executed in lexicographic order of their names, e.g.,
+	 * <code>test1</code> before <code>test2</code>. The name of this method is
+	 * chosen so that it comes lexicographically rather safely after all other
+	 * actual test methods.
+	 */
+	public void test___cleanup() throws Exception {
+		super.tearDown();
+	}
```

Comment:
This is an implementation detail, you shouldn't rely on that. Annotate the class with `@FixMethodOrder(MethodSorters.NAME_ASCENDING)` to make sure this holds.

Reply:
`@FixMethodOrder` will not have any effect here, since `WorkspaceSessionTests` are executed via a `WorkbenchSessionTestSuite` that simply sorts methods by their names (according to JUnit 3 behavior). This is to be improved when migrating to JUnit 4 and adapting the suite implementation for that.

Reply:
Good point, this is the preliminary work so these tests are still JUnit3.
Then simply keep this in mind when migrating to JUnit4 and use the annotation if possible :-)